magento/magento2
# Fix gallery rendering on page load so current breakpoint configuration isn't replaced by any other configuration

### Description (*)
Fixes:  #29933 

If we have custom configuration for more than one defined breakpoint in `etc/view.xml` current gallery js module triggers matchMedia method for every breakpoint. This method triggers entry or exit function on page load for every defined breakpoint so it can replace previous configuration. For example we load page on mobile and it first runs entry function for current breakpoint but will also run exit function for desktop breakpoint if defined. This way we can have wrong looking gallery.


### Fixed Issues (if relevant)

1. Fixes magento/magento2#29933

### Manual testing scenarios (*)
1. Set in your theme (etc/view.xml)

On default Luma settings we won't see this issue because configuration has only one breakpoint defined.
`"breakpoints": {"mobile":{"conditions":{"max-width":"767px"},"options":{"options":{"nav":"dots"}}}}            }`

So we need have this:

```xml
var name="breakpoints">
            <var name="desktop">
                <var name="conditions">
                    <var name="min-width">1024px</var>
                </var>
                <var name="options">
                    <var name="options">
                        <var name="navdir">vertical</var>
                    </var>
                </var>
            </var>
            <var name="mobile">
                <var name="conditions">
                    <var name="max-width">767px</var>
                </var>
                <var name="options">
                    <var name="options">
                        <var name="nav">dots</var>
                    </var>
                </var>
            </var>
        </var>
    </vars>
```

2. Load product page

3. On page load on mobile dot nav should be visible

### Questions or comments
This is quick fix (limited time). Perhaps it could be fixed in other way. Please verify. If you will notice better option let me know I will try to check it again.
